sktime/classification/signature_based/_signature_classifier.py

```diff
@@ -0,0 +1,140 @@
+    Generalised Signature Method for Time-Series":
+        [https://arxiv.org/pdf/2006.00873.pdf]
+
+    Note that the final classifier used on the UEA datasets involved tuning
+    the hyper-parameters:
+        - `depth` over [1, 2, 3, 4, 5, 6]
+        - `window_depth` over [2, 3, 4]
+        - RandomForestClassifier hyper-parameters.
+    as these were found to be the most dataset dependent hyper-parameters.
+
+    Thus, we recommend always tuning *at least* these parameters to any given
+    dataset.
+
+    Parameters
+    ----------
+    classifier: sklearn estimator, This should be any sklearn-type
+        estimator. Defaults to RandomForestClassifier.
+    augmentation_list: list of tuple of strings, List of augmentations to be
+        applied before the signature transform is applied.
+    window_name: str, The name of the window transform to apply.
+    window_depth: int, The depth of the dyadic window. (Active only if
+        `window_name == 'dyadic']`.
+    window_length: int, The length of the sliding/expanding window. (Active
+        only if `window_name in ['sliding, 'expanding'].
+    window_step: int, The step of the sliding/expanding window. (Active
+        only if `window_name in ['sliding, 'expanding'].
+    rescaling: str, The method of signature rescaling.
+    sig_tfm: str, String to specify the type of signature transform. One of:
+        ['signature', 'logsignature']).
+    depth: int, Signature truncation depth.
+    random_state: int, Random state initialisation.
+
+    Attributes
+    ----------------
+    signature_method: sklearn.Pipeline, An sklearn pipeline that performs the
+        signature feature extraction step.
+    pipeline: sklearn.Pipeline, The classifier appended to the
+        `signature_method` pipeline to make a classification pipeline.
+    """
+
+    def __init__(
+        self,
+        classifier=None,
+        augmentation_list=("basepoint", "addtime"),
+        window_name="dyadic",
+        window_depth=3,
+        window_length=None,
+        window_step=None,
+        rescaling=None,
+        sig_tfm="signature",
+        depth=4,
+        random_state=None,
+    ):
+        super(SignatureClassifier, self).__init__()
+        self.classifier = classifier
+        self.augmentation_list = augmentation_list
+        self.window_name = window_name
+        self.window_depth = window_depth
+        self.window_length = window_length
+        self.window_step = window_step
+        self.rescaling = rescaling
+        self.sig_tfm = sig_tfm
+        self.depth = depth
+        self.random_state = random_state
+        np.random.seed(random_state)
+
+        self.signature_method = GeneralisedSignatureMethod(
+            augmentation_list,
+            window_name,
+            window_depth,
+            window_length,
+            window_step,
+            rescaling,
+            sig_tfm,
+            depth,
+        ).signature_method
+
+    def setup_classification_pipeline(self):
```

Comment:
I'd suggest to make this private
```suggestion
    def _setup_classification_pipeline(self):
```